node/pkg/driver/node.go

```diff
@@ -673,23 +674,24 @@ func (d *NodeService) getVolumeStats(path string, volumeId string) (VolumeStatis
 	}
 
 	if isBlock {
-		volumeUuid := d.NodeUtils.GetVolumeUuid(volumeId)
-		mpathDevice, err := d.OsDeviceConnectivityHelper.GetMpathDevice(volumeUuid)
+		volumeStats, err = d.NodeUtils.GetBlockVolumeStats(volumeId)
 		if err != nil {
-			switch err.(type) {
-			case *device_connectivity.MultipathDeviceNotFoundForVolumeError:
-				return VolumeStatistics{}, status.Errorf(codes.NotFound, "Multipath device of volume id %q does not exist", volumeId)
-			default:
-				return VolumeStatistics{}, status.Errorf(codes.Internal, "Error while discovering the device : %s", err)
-			}
+			return VolumeStatistics{}, status.Errorf(codes.Internal, "Failed to get statistics: %s", err)
```

Comment:
does this mean we won't return `NotFound` status code in [this case](https://github.com/IBM/ibm-block-csi-driver/blob/v1.9.0/node/pkg/driver/node.go#L683) anymore?

Reply:
currently, you are right, do you want that I will add if statement there to handle this type of errors?

Reply:
we [already have a "switch-case"](https://github.com/IBM/ibm-block-csi-driver/blob/7edffb8246e0c770aa072002b74daefb0308cd09/node/pkg/driver/node_utils.go#L573-L578) for this (which is similar to an `if`).
IMO we should just keep this "switch-case" in `node.go`, so that `node_utils.go` won't be aware of CSI spec status codes.
i.e. `GetBlockVolumeStats` should return the error as is (whether it's `MultipathDeviceNotFoundForVolumeError` or not)

Reply:
could you also cover this in the unit tests?

Reply:
done

Reply:
are you sure it's covered? I checked out the branch and changed [this](https://github.com/IBM/ibm-block-csi-driver/blob/f72c7caf45c542bc7ed6eb4f35bc5641aa6c80f6/node/pkg/driver/node.go#L681) to `codes.Internal` and all the tests still passed

Reply:
done

Reply:
gentle reminder